### DISCUSSION TITLE
Unable to connect database

### DISCUSSION THREAD

**Discussion by qiaoj427** [atlarge-research/opendc#159] [General]
Hi, I have a question. I can not create my project or find any project existed after starting sever. It seems like that the database connection is wrong. 
There is a notification in the http://localhost:8080.
`Unable to connect
There was an error retrieving data. Check your connection and try again.`

There are some warnings in the log.
`2023-10-20 02:33:23,140 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-0,5,main] has been blocked for 2843 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
opendc-30-rc1_server_1: 	at java.base@17.0.2/sun.nio.ch.EPoll.wait(Native Method)
opendc-30-rc1_server_1: 	at java.base@17.0.2/sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:118)
opendc-30-rc1_server_1: 	at java.base@17.0.2/sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:129)
opendc-30-rc1_server_1: 	at java.base@17.0.2/sun.nio.ch.SelectorImpl.select(SelectorImpl.java:141)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:883)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:526)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
opendc-30-rc1_server_1: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.Thread.run(Thread.java:833)
opendc-30-rc1_server_1: 
opendc-30-rc1_server_1: 2023-10-20 02:33:36,625 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-0,5,main] has been blocked for 12484 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint.checkExpired(SharedClientHttpStreamEndpoint.java:115)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl.lambda$static$1(HttpClientImpl.java:111)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl$$Lambda$415/0x0000000801049f30.accept(Unknown Source)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4780)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.ConnectionManager.forEach(ConnectionManager.java:33)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl.checkExpired(HttpClientImpl.java:211)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl.access$100(HttpClientImpl.java:53)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl$PoolChecker.handle(HttpClientImpl.java:205)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl$PoolChecker.handle(HttpClientImpl.java:193)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:922)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:893)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.ContextInternal.emit(ContextInternal.java:194)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:911)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
opendc-30-rc1_server_1: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.Thread.run(Thread.java:833)
opendc-30-rc1_server_1: 
opendc-30-rc1_server_1: 2023-10-20 02:33:36,626 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-1,5,main] has been blocked for 13465 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass1(Native Method)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
opendc-30-rc1_server_1: 	at app//io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:105)
opendc-30-rc1_server_1: 	at app//io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:65)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass1(Native Method)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
opendc-30-rc1_server_1: 	at app//io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:105)
opendc-30-rc1_server_1: 	at app//io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:65)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass1(Native Method)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
opendc-30-rc1_server_1: 	at app//io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:105)
opendc-30-rc1_server_1: 	at app//io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:65)
opendc-30-rc1_server_1: 	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:113)
opendc-30-rc1_server_1: 	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:102)
opendc-30-rc1_server_1: 	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:95)
opendc-30-rc1_server_1: 	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:88)
opendc-30-rc1_server_1: 	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:81)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.transport.Transport$$Lambda$476/0x00000008010aeed8.newChannel(Unknown Source)
opendc-30-rc1_server_1: 	at io.netty.bootstrap.AbstractBootstrap.initAndRegister(AbstractBootstrap.java:310)
opendc-30-rc1_server_1: 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:155)
opendc-30-rc1_server_1: 	at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:139)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.ChannelProvider.handleConnect(ChannelProvider.java:155)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.ChannelProvider.connect(ChannelProvider.java:106)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.ChannelProvider.connect(ChannelProvider.java:92)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl.connectInternal2(NetClientImpl.java:273)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl.lambda$connectInternal$1(NetClientImpl.java:241)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl$$Lambda$472/0x00000008010a5138.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureBase$$Lambda$473/0x00000008010a5570.run(Unknown Source)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
opendc-30-rc1_server_1: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.Thread.run(Thread.java:833)
opendc-30-rc1_server_1: 
opendc-30-rc1_server_1: 2023-10-20 02:34:01,701 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-1,5,main] has been blocked for 22143 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.ClassLoader.defineClass0(Native Method)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.System$2.defineClass(System.java:2307)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.MethodHandles$Lookup$ClassDefiner.defineClass(MethodHandles.java:2439)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.MethodHandles$Lookup$ClassDefiner.defineClassAsLookup(MethodHandles.java:2420)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.MethodHandles$Lookup.defineHiddenClass(MethodHandles.java:2127)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.InnerClassLambdaMetafactory.generateInnerClass(InnerClassLambdaMetafactory.java:407)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.InnerClassLambdaMetafactory.spinInnerClass(InnerClassLambdaMetafactory.java:315)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.InnerClassLambdaMetafactory.buildCallSite(InnerClassLambdaMetafactory.java:228)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.LambdaMetafactory.metafactory(LambdaMetafactory.java:341)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.LambdaForm$DMH/0x0000000800e14000.invokeStatic(LambdaForm$DMH)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.Invokers$Holder.invokeExact_MT(Invokers$Holder)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:134)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.CallSite.makeSite(CallSite.java:315)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:281)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:271)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientRequestBase.<init>(HttpClientRequestBase.java:64)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientRequestImpl.<init>(HttpClientRequestImpl.java:62)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl.lambda$null$5(HttpClientImpl.java:641)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl$$Lambda$530/0x00000008011965d0.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl.addListener(FutureImpl.java:196)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.PromiseImpl.addListener(PromiseImpl.java:23)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl.onComplete(FutureImpl.java:164)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.PromiseImpl.onComplete(PromiseImpl.java:23)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.Http1xClientConnection.createStream(Http1xClientConnection.java:1199)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl.lambda$doRequest$6(HttpClientImpl.java:635)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.HttpClientImpl$$Lambda$460/0x0000000801099380.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.Endpoint.lambda$getConnection$0(Endpoint.java:52)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.Endpoint$$Lambda$466/0x000000080109e8e8.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:162)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:123)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.SimpleConnectionPool$LeaseImpl.emit(SimpleConnectionPool.java:704)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.SimpleConnectionPool$ConnectSuccess$2.run(SimpleConnectionPool.java:338)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:50)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:245)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.SimpleConnectionPool.lambda$connect$2(SimpleConnectionPool.java:257)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.pool.SimpleConnectionPool$$Lambda$467/0x000000080109d218.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint.lambda$connect$2(SharedClientHttpStreamEndpoint.java:102)
opendc-30-rc1_server_1: 	at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$$Lambda$468/0x000000080109d440.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl.addListener(FutureImpl.java:196)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.PromiseImpl.addListener(PromiseImpl.java:23)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
opendc-30-rc1_server_1: 	at io.vertx.core.Promise.complete(Promise.java:66)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl.lambda$connected$8(NetClientImpl.java:311)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl$$Lambda$517/0x0000000801182000.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.VertxHandler.setConnection(VertxHandler.java:82)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.VertxHandler.handlerAdded(VertxHandler.java:88)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.callHandlerAdded(AbstractChannelHandlerContext.java:938)
opendc-30-rc1_server_1: 	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:609)
opendc-30-rc1_server_1: 	at io.netty.channel.DefaultChannelPipeline.addLast(DefaultChannelPipeline.java:223)
opendc-30-rc1_server_1: 	at io.netty.channel.DefaultChannelPipeline.addLast(DefaultChannelPipeline.java:195)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl.connected(NetClientImpl.java:313)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl.lambda$connectInternal2$2(NetClientImpl.java:271)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.NetClientImpl$$Lambda$475/0x00000008010a3160.handle(Unknown Source)
opendc-30-rc1_server_1: 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
opendc-30-rc1_server_1: 	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:126)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:324)
opendc-30-rc1_server_1: 	at io.netty.handler.ssl.SslHandler.setHandshakeSuccess(SslHandler.java:1838)
opendc-30-rc1_server_1: 	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1360)
opendc-30-rc1_server_1: 	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1236)
opendc-30-rc1_server_1: 	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1285)
opendc-30-rc1_server_1: 	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:519)
opendc-30-rc1_server_1: 	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:458)
opendc-30-rc1_server_1: 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
opendc-30-rc1_server_1: 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
opendc-30-rc1_server_1: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
opendc-30-rc1_server_1: 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
opendc-30-rc1_server_1: 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
opendc-30-rc1_server_1: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
opendc-30-rc1_server_1: 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
opendc-30-rc1_server_1: 	at java.base@17.0.2/java.lang.Thread.run(Thread.java:833)
opendc-30-rc1_server_1: 
opendc-30-rc1_server_1: 2023-10-20 02:34:01,721 WARN  [io.qua.oid.com.run.OidcCommonUtils] (vert.x-eventloop-thread-1) OIDC Server is not available:: io.vertx.core.http.HttpClosedException: Connection was closed`
Are there any configuration that I miss?
Your suggestion and help would be appreciated.

**Comment by fabianishere** [comment]
What's the command you used to launch OpenDC? Could you provide the full logs if possible?

**Reply by qiaoj427** [reply]
Thanks for your reply. I have changed the version from v3.0 to v2.0. Now I have completed 'docker-compose build' progress, and I successfully run the command 'docker-compose up'. After that, I open the webpage, and there is a new problem as followed.
`Welcome to nginx!
If you see this page, the nginx web server is successfully installed and working. Further configuration is required.

For online documentation and support please refer to [nginx.org](http://nginx.org/).
Commercial support is available at [nginx.com](http://nginx.com/).

Thank you for using nginx.`
So what should I do now?
